opm/grid/cpgrid/CpGridData.cpp

```diff
@@ -2204,13 +2204,21 @@ bool CpGridData::hasNNCs(const std::vector<int>& cellIndices) const
 void CpGridData::validStartEndIJKs(const std::vector<std::array<int,3>>& startIJK_vec,
                                    const std::vector<std::array<int,3>>& endIJK_vec) const
 {
-    assert(startIJK_vec.size() == endIJK_vec.size());
-    for (unsigned int patch = 0; patch < startIJK_vec.size(); ++patch) {
-        for (int c = 0; c < 3; ++c) {
-            // valid startIJK and endIJK for each patch
-            assert(startIJK_vec[patch][c] < endIJK_vec[patch][c]);
+    if (startIJK_vec.size() == endIJK_vec.size()) {
+        for (unsigned int patch = 0; patch < startIJK_vec.size(); ++patch) {
+            bool validPatch = true;
+            for (int c = 0; c < 3; ++c) {
+                // valid startIJK and endIJK for each patch
+                validPatch = validPatch && (startIJK_vec[patch][c] < endIJK_vec[patch][c]);
+                if (!validPatch) {
+                    OPM_THROW(std::invalid_argument, "Invalid ijk-indices, e.g. end-i smaller than start-i.");
```

Comment:
```suggestion
                    OPM_THROW(std::invalid_argument, "Invalid ijk-indices. end[" std::to_string(c) "] not larger than start[" std::to_string(c) "]");
```
Not 100% sure whether this suggestion actually compiles. I did not understand the i in the original message

Reply:
I meant to say that at least one of the coordinates is invalid, meaning that, if startIJK = {i1, j1, k1} and endIJK={ i2,j2,k2}, then, i1>=i2. I agree, it was not clear. Thanks for the suggestions!